cumulusci/tasks/bulkdata.py

```diff
@@ -797,6 +799,313 @@ def _sqlite_dump(self):
+                "(Deprecated)" in field["label"],  # Deprecated managed fields
+                field["type"] == "base64",  # No Bulk API support for base64 blob fields
+                not field["createable"],  # Non-writeable fields
+                field["type"] == "reference"  # Self-lookups
+                and field["referenceTo"] == [obj],
+                field["type"] == "reference"  # Outside lookups
+                and not self._are_lookup_targets_in_operation(field),
+            ]
+        )
+
+    def _is_required_field(self, field):
+        return (field["createable"] and not field["nillable"]) or (
+            field["type"] == "reference" and field["relationshipOrder"] == 1
+        )
+
+    def _has_our_custom_fields(self, obj):
+        return any(
+            [self._is_our_custom_api_name(field["name"]) for field in obj["fields"]]
+        )
+
+    def _are_lookup_targets_in_operation(self, field):
+        return all([f in self.mapping_objects for f in field["referenceTo"]])
+
+    def _is_lookup_to_included_object(self, field):
+        return field["type"] == "reference" and self._are_lookup_targets_in_operation(
+            field
+        )
+
+    def _collect_objects(self):
+        self.mapping_objects = []
+
+        # Cache the global describe, which we'll walk.
+        self.global_describe = self.sf.describe()
+
+        # First, we'll get a list of all objects that are either
+        # (a) custom, no namespace
+        # (b) custom, with our namespace
+        # (c) not ours (standard or other package), but have fields with our namespace or no namespace
+        self.describes = {}  # Cache per-object describes for efficiency
+        for obj in self.global_describe["sobjects"]:
+            self.describes[obj["name"]] = getattr(self.sf, obj["name"]).describe()
+            if self._is_our_custom_api_name(obj["name"]) or self._has_our_custom_fields(
+                self.describes[obj["name"]]
+            ):
+                if self._is_object_mappable(obj):
+                    self.mapping_objects.append(obj["name"])
+
+        # Add any objects that are required by our own,
+        # meaning any object we are looking up to with a custom field,
+        # or any master-detail parent of any included object.
+        index = 0
+        while index < len(self.mapping_objects):
+            obj = self.mapping_objects[index]
+            for field in self.describes[obj]["fields"]:
+                if field["type"] == "reference":
+                    if field["relationshipOrder"] == 1 or self._is_any_custom_api_name(
+                        field["name"]
+                    ):
+                        self.mapping_objects.extend(
+                            filter(
+                                lambda o: o not in self.mapping_objects
+                                and self._is_object_mappable(self.describes[o]),
+                                field["referenceTo"],
+                            )
```

Comment:
How about:

    self.mapping_objects.extend([obj for obj in field['referenceTo'] if obj not in self.mapping_objects and self._is_object_mappable(self.describes[obj])])

Maybe this is just taste but I almost find comprehensions more readable than filter or map with a lambda expression.